queries/nim/highlights.scm

```diff
@@ -0,0 +1,797 @@
+; @method.call      ; method calls
+; unused
+
+; =============================================================================
+; @constructor      ; constructor calls and definitions
+
+(call
+  function: [
+    (identifier) @constructor
+    (accent_quoted (identifier) @constructor)
+    ; generic types
+    (bracket_expression left: (identifier) @constructor)
+    (bracket_expression left: (accent_quoted (identifier) @constructor))
+    ; dot accessor
+    (dot_expression right: (identifier) @constructor)
+    (dot_expression right: (accent_quoted (identifier) @constructor))
+    ; both
+    (bracket_expression left:
+      (dot_expression right: (identifier) @constructor))
+    (bracket_expression left:
+      (dot_expression right: (accent_quoted (identifier) @constructor)))
+  ]
+  (argument_list
+    (colon_expression)+))
+; NOTE: this cannot detect constructors with 0 arguments
+; those will be matched as @function.call instead
+
+; =============================================================================
+; @parameter        ; parameters of a function
+
+; named parameters when calling
+; call(parameter_name=arg)
+(argument_list
+  (equal_expression
+    left: [
+      (identifier) @parameter
+      (accent_quoted (identifier) @parameter)
+    ]))
+
+; parameters in function declaration
+(parameter_declaration_list
+  (parameter_declaration
+    (symbol_declaration_list
+      (symbol_declaration
+        name: [
+          (identifier) @parameter
+          (accent_quoted (identifier) @parameter)
+        ]))))
+
+; for loop variables
+(for
+  left:
+    (symbol_declaration_list
+      (symbol_declaration
+        name: [
+          (identifier) @parameter
+          (accent_quoted (identifier) @parameter)
+        ])))
+
+((tuple_deconstruct_declaration
+  (symbol_declaration
+    name: [
+      (identifier) @parameter
+      (accent_quoted (identifier) @parameter)
+    ])) @_tuple_decons
+  (#has-ancestor? @_tuple_decons for))
+
+; =============================================================================
+; @keyword             ; various keywords
+
+; unhandled but reserved keywords
+; end
+; interface
+
+; static expression
+; addr operator
+((call
+  function: (identifier) @keyword)
+  (#any-of? @keyword "static" "addr"))
+
+[
+  "const"
+  "let"
+  "var"
+  "type"
+  "concept"
+  "asm"
+  "bind"
+  "defer"
+  "do"
+  "mixin"
+  "static"
+  "object"
+  "tuple"
+  "enum"
+  "block"
+  "using"
+  "discard"
+] @keyword
+
+; =============================================================================
+; @keyword.coroutine   ; keywords related to coroutines (e.g. `go` in Go, `async/await` in Python)
+; unused
+
+; =============================================================================
+; @keyword.function    ; keywords that define a function (e.g. `func` in Go, `def` in Python)
+
+[
+  "proc"
+  "func"
+  "method"
+  "converter"
+  "iterator"
+  "macro"
+  "template"
```

Comment:
Maybe `@preproc` for these two?

Reply:
doesn't even make a difference for my theme.

`macro` and `template` share the syntax with the other function keywords though and I am already capturing the identifier following `macro` and `template` keywords as `@function.macro` and the keywords don't appear standalone.

Reply:
The colour is irrelevant. If these define preprocessor functions, then `@preproc` is appropriate.